turbo/stages/headerdownload/header_algos.go

```diff
@@ -682,6 +719,75 @@ func (hd *HeaderDownload) InsertHeaders(hf func(header *types.Header, headerRaw
 	return hd.highestInDb >= hd.preverifiedHeight && hd.topSeenHeight > 0 && hd.highestInDb >= hd.topSeenHeight, nil
 }
 
+func (hd *HeaderDownload) InsertHeadersBackwards(tx kv.RwTx, canonicalHashCollector *etl.Collector, headerCollector *etl.Collector, logPrefix string, logChannel <-chan time.Time) (bool, error) {
+	if len(hd.insertList) == 0 {
+		return false, nil
+	}
+	hd.lock.Lock()
+	defer hd.lock.Unlock()
+	sort.Slice(hd.insertList, func(i, j int) bool {
+		return hd.insertList[i].header.Number.Uint64() > hd.insertList[j].header.Number.Uint64()
+	})
+	for len(hd.insertList) > 0 {
```

Comment:
It is probably better to just iterate over the `insertList` and make it empty after the loop than to do re-slicing

Reply:
I mean this:
```
for _, header := range hd.insertList {
...
}
hd.insertList = hd.insertList[:0]
```
it is more efficient and more readable that way